### PR TITLE
Refactor rule evaluation handling and add utility function for formatting rule evaluation status

### PR DESCRIPTION
**Pull request by yeswanth2420**
This pull request refactors how the `ruleEval` field is handled in the feature control handler by standardizing its format. Instead of sometimes storing a string and sometimes a string slice, a new utility function ensures `ruleEval` is always a formatted string. This improves consistency and simplifies downstream processing.

**Refactoring and Consistency Improvements:**

* Introduced `util.FormatRuleEvalStatus`, a utility function to consistently format `ruleEval` as a string, joining multiple reasons with a dash if necessary (`util/string.go`).
* Updated assignments to the `fields["ruleEval"]` field in `GetFeatureControlSettingsHandler` to use the new formatting function, ensuring `ruleEval` is always a string rather than sometimes a slice (`dataapi/feature_control_handler.go`). [[1]](diffhunk://#diff-7f30d83d14c56846765c3e856d7a7625cd76895d753a193fae92783e337a7f97L194-R194) [[2]](diffhunk://#diff-7f30d83d14c56846765c3e856d7a7625cd76895d753a193fae92783e337a7f97L277-R277)

### CODE REVIEW

File: util/string.go [hunk 1]
```diff
@@ -231,6 +231,13 @@ func IsBlank(str string) bool {
 	return strings.Trim(str, " ") == ""
 }
 
+func FormatRuleEvalStatus(ruleEvalReasons []string) string {
+	if len(ruleEvalReasons) == 1 {
+		return ruleEvalReasons[0]
+	}
+	return strings.Join(ruleEvalReasons, "-")
+}
```
[the remark this answers is not in the document]
Comment (yeswanth2420) [reply util/string.go:236]: Resolved

**Review by gravi21 — APPROVED** [review]

**Review by coolwater — COMMENTED** [review]
Refactor looks good—clean abstraction and solid test coverage. Utility simplifies formatting and keeps behavior consistent. Minor suggestions: consider adding nil input test, documenting empty slice behavior, and clarifying naming (“status” vs “reasons”). Otherwise LGTM.
